### PR TITLE
Caching

### LINKED ISSUE #108: Caching

### PR DESCRIPTION
**Pull request by domo2192**
----MASSSIVEEE SPEED BOOOSTS 

Places I added caching 
-api calls for purposes on the front end 
- api calls for yard details on the front end 
- some partial rendering 
- yard_details cache is being deleted whenever yards are updated 

Testing
Ian said testing is for chumps but all our tests are passing right now

### CODE REVIEW

File: app/services/engine_service.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 class EngineService
 
   def self.all_purposes
-    response = connection.get('/api/v1/purposes')
-    JSON.parse(response.body, symbolize_names: true)
+    Rails.cache.fetch("all_purposes", expires_in: 24.hours) do
+      response = connection.get('/api/v1/purposes')
```
Comment (amsmyth1) [inline app/services/engine_service.rb:5]: this is really exciting to see implemented!

File: app/services/engine_service.rb [hunk 2]
```diff
@@ -15,17 +17,21 @@ def self.create_yard(yard_params)
   end
 
   def self.update_yard(yard_params)
-    response = connection.put("/api/v1/yards/#{yard_params[:id]}") do |req|
-      req.headers["CONTENT_TYPE"] = "application/json"
-      req.params = yard_params
+    Rails.cache.delete(["yard", yard_params[:id]])
+      response = connection.put("/api/v1/yards/#{yard_params[:id]}") do |req|
```
Comment (amsmyth1) [inline app/services/engine_service.rb:21]: nice! love that rails magic.

**Review by amsmyth1 — APPROVED** [review]
This is awesome! Thanks for tackling this new concept

**Review by abreaux26 — APPROVED** [review]
💪🏼 HUGE WIN

**Pull request merged**